pgmax/fg/graph.py

```diff
@@ -853,24 +854,34 @@ def tree_unflatten(cls, aux_data, children):
         return cls(**aux_data.unflatten(children))
 
 
-def BP(
-    bp_state: BPState, num_iters: int, temperature: float = 0.0
-) -> Tuple[Callable, Callable, Callable]:
+class BPContainer(NamedTuple):
```

Comment:
Container sounds weird...How about just `BeliefPropagation`? Also shorten all the existing `bp_container` to just `bp`.

Reply:
Our internal software criteria says use dataclasses not named tuples. Follow those and use dataclasses for now?